package/base-files/files/lib/upgrade/do_stage2

```diff
@@ -18,6 +18,11 @@ fi
 v "Upgrade completed"
 sleep 1
 
+if is_efi_system; then
+	ask_bool 0 "Reboot now?" && reboot
+	return 0
+fi
```

Comment:
Why should this be different between EFI and non-EFI systems?  I can understand wanting to ask before rebooting, but why not also ask on non-EFI systems?